moddingway/services/note_service.py

```diff
@@ -0,0 +1,161 @@
+import discord
+import logging
+from database import users_database
+from util import log_info_and_embed, log_info_and_add_field
+from database import notes_database, users_database
+from database.models import Note
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+
+async def add_note(
+    logging_embed: discord.Embed,
+    user: discord.Member,
+    note: str,
+    author: discord.Member,
+):
+    # find user in DB
+    db_user = users_database.get_user(user.id)
+    if db_user is None:
+        log_info_and_embed(
+            logging_embed,
+            logger,
+            "User not found in database, creating new record",
+        )
+        db_user = users_database.add_user(user.id)
```

Comment:
(mostly commenting for myself) This pattern has started to appear enough in the codebase that it should be move to some sort of method or other service. You shouldn't do that as part of this PR tho